### PR TITLE
full transfer learning guide

### PR DESCRIPTION
**Pull request by cgarciae**
# What does this PR do?

Expanded version of the Transfer Learning guide with a complete example.

### CODE REVIEW

File: docs/guides/transfer_learning_full.md [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+kernelspec:
+  display_name: 'Python 3.8.11 (''.venv'': venv)'
+  language: python
+  name: python3
+---
+
+# Transfer Learning
+
+
+### Ideas
```
Comment (marcvanzee) [inline docs/guides/transfer_learning_full.md:18]: Is this meant to be part of the final version? It looks like a bit of an odd section to start with.

File: docs/guides/transfer_learning_full.md [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+* Dataset: use a simple dataset from TensorFlow Dataset like MNIST, KMNIST, or FashionMNIST, or even cats vs dogs which is a little bigger.
+* Model Options: 
+  * `jax-resnet`: has the benefit of being small and fast.
+  * `FlaxCLIPVisionModel`: would showcase how to use HuggingFace models.
+
+### Important Links
+- https://github.com/google/flax/discussions/1706
+- https://github.com/google/flax/discussions/2245
+
+### Outline
```
Comment (marcvanzee) [inline docs/guides/transfer_learning_full.md:28]: I wouldn't add this to the guide either.

File: docs/guides/transfer_learning_full.md [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+    * Completely freezing layers
+    * Using different learning rates
+* Training the model
+
++++
+
+## Introduction
+
+### What is Transfer Learning?
+Theory
```
Comment (marcvanzee) [inline docs/guides/transfer_learning_full.md:49]: I wouldn't use bullets here, also the separation between "theory" and "practice" is a bit arbitrary, why is "Pretrain on a base task, use the learned parameters to train on a target task" theory and "Fine-tuning: use the pretrained weights as a starting point but train the whole model on the new task." practice?

Please also add some sources, could be articles but also (good) blog posts where concepts of transfer learning are discussed.

File: docs/guides/transfer_learning_full.md [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+* Fine-tuning: use the pretrained weights as a starting point but train the whole model on the new task. This is commonly done in NLP tasks.
+* Extract features: use the pretrained model to extract the features for the target task and train a new model on those features. This is commonly done in vision tasks. (mention pre-computing all embeddings as an optimization)
+* Hybrid: you can interpolate between the two previous approaches by deciding how much of the pretrained model remains frozen and how much is trained, e.g. freeze the last few layers and train the rest. Apart from that you can also interpolate between freezing and training by using different learning rates per layer where freezing is just the extreme case of the learning rate being 0.
+
+### Project Description
+For this tutorial we will be creating a model for a simple Computer Vision classification task on a small dataset, more specifically we will using a pretrained `MlpMixer` Flax model from the [vision_transformer](https://github.com/google-research/vision_transformer) repository to learn the [rock_paper_scissors](https://www.tensorflow.org/datasets/catalog/rock_paper_scissors) dataset from TensorFlow Datasets.
```
Comment (marcvanzee) [inline docs/guides/transfer_learning_full.md:61]: This seems like a good sentence to start the guide with.

File: docs/guides/transfer_learning_full.md [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+The first thing you need to do is load a pretrained, depending on which model/library you are using this might look wildly different but in general you will have to get hold of the following:
+
+* The Flax `Module` that defines the model architecture.
+* The model parameters (and batch statistics for models that use batch norm).
+* The preprocessing procedure that converts the raw inputs into the format the model expects, for images this is often a simple processing scripts that applies resize and some normalization, and for text this usually a tokenizer.
+
+In this tutorial we will create an `MlpMixer` model using the `vit_jax` library from the `vision_transformer` repository. The thing we need to do is to install the `vit_jax` library:
+
+```{code-cell} ipython3
+! pip install transformers[flax]
```
Comment (marcvanzee) [inline docs/guides/transfer_learning_full.md:73]:
```suggestion
!pip install transformers[flax]
```

**Review by marcvanzee — CHANGES_REQUESTED** [review]

File: docs/guides/transfer_learning_full.md (hunk 5 above)
Comment (8bitmp3) [reply docs/guides/transfer_learning_full.md:73]: Also `!pip install -U flax jax jaxlib` since `transformers` uses different versions of Flax and JAX.

**Comment by cgarciae** [comment]
Maybe we can close it for now, in the future maybe this could turn into a Tutorial.

**Pull request closed without merging**